pkg/discovery/discovery.go

```diff
@@ -124,7 +128,55 @@ func Run(kubeClient kubernetes.Interface, cfg *config.Config) (errCount int) {
 		defer os.RemoveAll(outpath)
 	}
 	trackErrorsFor("assembling results tarball")(err)
+
+	// 9. Mark final annotation stating the results are available and status is completed.
+	trackErrorsFor("updating pod status")(
+		updateStatus(kubeClient, cfg.Namespace, pluginaggregation.CompleteStatus),
+	)
+
 	logrus.Infof("Results available at %v", tb)
 
 	return errCount
 }
+
+// updateStatus changes the summary status of the sonobuoy pod in order to
+// effect the finalized status the user sees. This does not change the status
+// of individual plugins.
+func updateStatus(client kubernetes.Interface, namespace string, s string) error {
+	pod, err := client.CoreV1().Pods(namespace).Get(pluginaggregation.StatusPodName, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "could not retrieve sonobuoy pod")
+	}
+
+	if pod.Status.Phase != corev1.PodRunning {
+		return fmt.Errorf("pod has status %q", pod.Status.Phase)
+	}
+
+	statusJSON, ok := pod.Annotations[pluginaggregation.StatusAnnotationName]
```

Comment:
nit: It might make this method a bit cleaner to break out the `getting the status from annotations` and `parsing status` from the `updateStatus` method. They might be re-usable elsewhere if we're getting/setting status in other places.

Reply:
👍 In the next tweak I moved all the code actually out of the `pkg/client` and into the `plugins/aggregation` since all the types and constants in question were defined there. Used in both the `client` and `discovery` packages now.